app/controllers/budgets_controller.rb

```diff
@@ -41,10 +46,10 @@ def destroy
   private
 
   def set_budget
-    @budget = current_user.budgets.find(params[:id])
+    @budget = current_user.budgets.find_by(id: params[:id])
```

Comment:
It will result into error if budget with given id is not found and you are going to update or destroy it.